src/step-function.ts

```diff
@@ -466,6 +466,428 @@ export namespace $SFN {
+            })`,
+          },
+          ResultPath: temp,
+          Next: ASLGraph.DeferNext,
+          output: {
+            jsonPath: `${temp}.out`,
+          },
+        };
+      });
+    },
+  });
+
+  /**
+   * The States.ArrayUnique intrinsic function removes duplicate values from an array and returns an array containing only unique elements.
+   * This function takes an array, which can be unsorted, as its sole argument.
+   *
+   * @param arr - array of values to return unique values of.
+   */
+  export const unique = makeStepFunctionIntegration<
+    "SFN.Unique",
+    <T>(arr: T[]) => T[]
+  >("SFN.Unique", {
+    asl(call, context) {
+      const [arr] = call.args;
+
+      if (!arr) {
+        throw new SynthError(
+          ErrorCodes.Invalid_Input,
+          "Expected Unique array arguments to be provided."
+        );
+      }
+
+      return context.evalContext(call, ({ evalExprToJsonPath }) => {
+        const arrayOut = evalExprToJsonPath(arr.expr);
+        const temp = context.newHeapVariable();
+        return {
+          Type: "Pass",
+          Parameters: {
+            "out.$": `States.ArrayUnique(${arrayOut.jsonPath})`,
+          },
+          ResultPath: temp,
+          Next: ASLGraph.DeferNext,
+          output: {
+            jsonPath: `${temp}.out`,
+          },
+        };
+      });
+    },
+  });
+
+  /**
+   * Use the `base64Encode` intrinsic function to encode data based on MIME Base64 encoding scheme.
+   * You can use this function to pass data to other AWS services without using an AWS Lambda function.
+   *
+   * @param data - String to encode as base64. Up to 10000 characters.
+   */
+  export const base64Encode = makeStepFunctionIntegration<
+    "SFN.Base64Encode",
+    (data: string) => string
+  >("SFN.Base64Encode", {
+    asl(call, context) {
+      const [data] = call.args;
+
+      if (!data) {
+        throw new SynthError(
+          ErrorCodes.Invalid_Input,
+          "Expected Base64 data argument to be provided."
+        );
+      }
+
+      return context.evalContext(call, ({ evalExprToJsonPathOrLiteral }) => {
+        const dataOut = evalExprToJsonPathOrLiteral(data.expr);
+
+        if (
+          ASLGraph.isLiteralValue(dataOut) &&
+          typeof dataOut.value !== "string"
+        ) {
+          throw new SynthError(
+            ErrorCodes.Invalid_Input,
+            "Expected base64Encode data argument to be a string or reference."
+          );
+        }
+        const temp = context.newHeapVariable();
+        return {
+          Type: "Pass",
+          Parameters: {
+            "out.$": `States.Base64Encode(${
+              ASLGraph.isJsonPath(dataOut)
+                ? dataOut.jsonPath
+                : `'${dataOut.value}'`
+            })`,
+          },
+          ResultPath: temp,
+          Next: ASLGraph.DeferNext,
+          output: {
+            jsonPath: `${temp}.out`,
+          },
+        };
+      });
+    },
+  });
```

Comment:
I wonder if there is anyway to simplify these new intrinsic. They seem to have a lot of code in common. Thoughts?

Reply:
Created a helper to assign a single parameter key and output it. Was applicable in many places.